R/mediationanalysis.R

```diff
@@ -340,26 +341,44 @@ MediationAnalysis <- function(jaspResults, dataset, options, ...) {
   dirtab[["ci.lower"]] <- pe_dir$ci.lower
   dirtab[["ci.upper"]] <- pe_dir$ci.upper
   dirtab$addFootnote(foot_message)
+
+  # Fill indirect effects
   pe_ind <- pe[pe$op == ":=" & vapply(gregexpr("_", pe$lhs), length, 1) == 3, ]
-  
-  indtab[["x"]]        <- rep(options$predictor, each = length(options$mediators) * length(options$dependent))
+  # get predictors, mediators, outcome combinations
+  terms <- strsplit(pe_ind[["label"]], "_")
+  termsCombinations <- list(
+    x = .medGetTermsFromLavaanTable(terms, 2, "x"),
+    m = .medGetTermsFromLavaanTable(terms, 3, "m"),
+    y = .medGetTermsFromLavaanTable(terms, 4, "y")
+  )
+
+  indtab[["x"]]        <- .unv(options[["predictor"]][termsCombinations[["x"]]])
```

Comment:
is `.unv` necessary here? This should be done by default since automatic column encoding was enabled.

Reply:
Yeah you are right...